internal/transport/http2_server.go

```diff
@@ -1345,3 +1345,18 @@ func getJitter(v time.Duration) time.Duration {
 	j := grpcrand.Int63n(2*r) - r
 	return time.Duration(j)
 }
+
+type connectionKey struct{}
+
+// GetConnection gets the connection from the context.
+func GetConnection(ctx context.Context) net.Conn {
+	conn, _ := ctx.Value(connectionKey{}).(net.Conn)
+	return conn
+}
+
+// SetConnection adds the connection to the context to be able to get
+// information about the destination ip and port for an incoming RPC. This also
+// allows any unary or streaming interceptors to see the connection.
+func SetConnection(ctx context.Context, conn net.Conn) context.Context {
```

Comment:
I get first one (did that with context passed from resolver to balancer), but I don't understand the second one at all.

Reply:
The second one is to do dependency injection where you have your test.  Instead of actually putting a conn into the context, fake the GetConnection function so it returns your test conn.  Just like how we mock time.

Reply:
Why would I want to inject my own connection here?  All I need to do is test whether the connection actually gets there, nothing to verify in the connection itself. If this somehow loses it's wrapping layer, it will get picked up in integration tests once I add routing to xds(Unary|Stream)Interceptors. GetConnection() will need to be called by rbac, so no need for GetConnectionForTesting. SetConnection makes sense, doing that now.

Reply:
> Why would I want to inject my own connection here? All I need to do is test whether the connection actually gets there, nothing to verify in the connection itself.

I'm talking about your unit tests of `rbac`.  `TestChainEngine` does `transport.SetConnection`.  Instead of that, it would be better to mock `transport.GetConnection` so you don't need to access `transport.SetConnection` -- only the transport should need that method.

Reply:
Oh I see what you're saying. Doing that now.

Reply:
Done.